### DISCUSSION TITLE
How can I get an accurate glyph dump for a particular font in SVG format? I've tried fonts2svg and FontForge (ffpython).

### DISCUSSION THREAD

**Discussion by futuremotiondev** [fonttools/fonttools#3158] [Q&A]
`fonts2svg HelveticaNeueLTPro-Bd.otf -o C:\Users\[username]\Desktop\Test\HelveticaNeueLTPro-Bd\SVG Dump`

This gives me a folder of SVGs, but some of the glyphs just show up as black boxes.

![explorer_EeXLXWe4Oy](https://github.com/fonttools/fonttools/assets/33441569/0c8346bc-9bc9-493d-b95d-05900a1b492a)

This is a little out of scope for font tools, but I've also tried FontForge from CLI:

`ffpython "D:\Dev\Python\FontSripts\ExportGlyphsToSVG.py" HelveticaNeueLTPro-Bd.otf "HelveticaNeueLTPro-Bd FontForge"`

```py
import fontforge
import os.path
import sys

if len(sys.argv) < 3:
    print("Usage: [FONT] [OUTPUTDIR]")
    exit(1)

fontpath = sys.argv[1]
font     = fontforge.open(fontpath)
outdir   = sys.argv[2]

if os.path.exists(outdir):
    pass
elif os.access(os.path.dirname(outdir), os.W_OK):
    os.makedirs(outdir)
else:
    exit(1)

for name in font:
    filename = name + ".svg"
    outfull = os.path.join(outdir,filename)
    font[name].export(outfull)
```

The results of the above command is even worse:

![explorer_ToiEIPDLlU](https://github.com/fonttools/fonttools/assets/33441569/2e71c9b0-ecfd-4062-9d2e-a5bd812f9039)

Is there a fool-proof method of extracting SVG glyphs from any given font that I am unaware of? (Preferably a command line option)

I would REALLY appreciate any input and alternative solutions.

Thanks

**Comment by justvanrossum** [comment]
If fonts2svg does what you need, apart from the black boxes, I suggest to file an issue with that tool. I bet they'll appreciate the feedback.
